daemon/cmd/datapath.go

```diff
@@ -275,6 +279,12 @@ func (d *Daemon) syncEndpointsAndHostIPs() error {
 		if err != nil {
 			return err
 		}
+		if option.Config.EnableL7Proxy {
+			err = setupRouteToVtepCidr()
```

Comment:
Minor nit here, only applying the route diff if the L7 proxy is enabled means that if you disable L7 proxy... the host routes will stay populated. However I think you intended that if L7 proxy is disabled, then all routes are removed? So probably this condition should wrap just the "addevVtepRoutes" case inside this function.